--- mstflint-4.10.0-3/mlxsign_lib/mlxsign_lib.cpp.orig	2018-08-28 15:36:51.000000000 +0200
+++ mstflint-4.10.0-3/mlxsign_lib/mlxsign_lib.cpp	2018-10-31 21:24:51.980647273 +0100
@@ -360,8 +360,7 @@
 
 MlxSignHMAC::MlxSignHMAC()
 {
-    ctx = malloc(sizeof(HMAC_CTX));
-    HMAC_CTX_init((HMAC_CTX*)ctx);
+    ctx = HMAC_CTX_new();
 }
 
 int MlxSignHMAC::setKey(const std::vector<u_int8_t>& key)
@@ -415,8 +414,7 @@
 
 MlxSignHMAC::~MlxSignHMAC()
 {
-    HMAC_CTX_cleanup((HMAC_CTX*)ctx);
-    free(ctx);
+    HMAC_CTX_free((HMAC_CTX*)ctx);
 }
 
 #endif //ENABLE_OPENSSL
