--- mstflint-4.31.0/tools_layouts/adb_to_c_utils.h.orig	2025-02-02 14:05:06.000000000 +0100
+++ mstflint-4.31.0/tools_layouts/adb_to_c_utils.h	2025-04-20 21:41:25.600451578 +0200
@@ -181,7 +181,7 @@ extern "C"
 #define U16H_FMT "0x%04x"
 #define U8H_FMT "0x%02x"
 
-#if defined(ARCH_x86) || defined(ARCH_ppc) || defined(UEFI_BUILD) || defined(ARCH_arm6l) || defined(ARCH_m68k) || defined(ARCH_hppa)
+#if defined(ARCH_x86) || defined(ARCH_ppc) || defined(UEFI_BUILD) || defined(ARCH_arm6l) || defined(ARCH_m68k) || defined(ARCH_hppa) || (defined(ARCH_x86_64) && defined(__ILP32__))
 #if defined(__MINGW32__) || defined(__MINGW64__)
 #include <inttypes.h>
 #define U64D_FMT "0x%" PRId64
--- mstflint-4.31.0/common/compatibility.h.orig	2025-02-02 14:05:06.000000000 +0100
+++ mstflint-4.31.0/common/compatibility.h	2025-04-20 21:45:17.029197820 +0200
@@ -41,7 +41,7 @@
 
 #include <stdio.h>
 
-#if defined(__ia64__) || defined(__x86_64__) || defined(__PPC64__) || defined(__arm__) || defined(__e2k__)
+#if defined(__ia64__) || (defined(__x86_64__) && !defined(__ILP32__)) || defined(__PPC64__) || defined(__arm__) || defined(__e2k__)
 #define U64L "l"
 #else
 #define U64L "ll"
@@ -104,7 +104,7 @@
 #define U16H_FMT "0x%04x"
 #define U8H_FMT "0x%02x"
 
-#if defined(ARCH_x86) || defined(ARCH_ppc) || defined(UEFI_BUILD) || defined(ARCH_arm6l) || defined(ARCH_m68k) || \
+#if defined(ARCH_x86) || (defined(ARCH_x86_64) && defined(__ILP32__)) || defined(ARCH_ppc) || defined(UEFI_BUILD) || defined(ARCH_arm6l) || defined(ARCH_m68k) || \
   defined(ARCH_hppa)
 #if defined(__MINGW32__) || defined(__MINGW64__)
 #include <inttypes.h>
